### PR TITLE
Renames the semantic versioning variable to CMT

### LINKED ISSUE #1200: Semantic version still references TM
TMCoreSemVer should be renamed to CMTSemVer.
And the version output by the binary should be named CometBFT, not Tendermint.

From the codebase, 

      grep -ri version.TMCoreSemVer .
      ./statesync/syncer_test.go:			Software: version.TMCoreSemVer,
      ./statesync/stateprovider.go:		Software:  version.TMCoreSemVer,
      ./cmd/cometbft/commands/version.go:		cmtVersion := version.TMCoreSemVer
      ./proxy/version.go:	Version:      version.TMCoreSemVer,
      ./test/e2e/README.md:# In the example below, if the local version.TMCoreSemVer value is "v0.34.24",
      ./test/e2e/generator/generate.go:	return findLatestReleaseTag(version.TMCoreSemVer, tags)
      ./config/config.go:		Version:            version.TMCoreSemVer,
      ./state/rollback.go:			Software: version.TMCoreSemVer,
      ./state/rollback_test.go:			Software:  version.TMCoreSemVer,
      ./state/rollback_test.go:			Software:  version.TMCoreSemVer,
      ./state/rollback_test.go:			Software: version.TMCoreSemVer,
      ./state/state.go:	Software: version.TMCoreSemVer,
      ./.goreleaser.yml:      - -s -w -X github.com/cometbft/cometbft/version.TMCoreSemVer={{ .Version }}
      ./node/setup.go:		"tendermint_version", version.TMCoreSemVer,
      ./node/node.go:		Version:       version.TMCoreSemVer,## Bug Report

When running

      2023-08-02 15:12:35 I[2023-08-02|18:12:35.314] Version info                                 module=main tendermint_version=0.39.0-dev abci=2.0.0 block=11 p2p=8 commit_hash=

### PR DESCRIPTION
**Pull request by lasarojc**
Closes #1200 

- [ ] Run a mock of the release process to ensure correct.

---

#### PR checklist

- [ ] Tests written/updated
- [ ] Changelog entry added in `.changelog` (we use [unclog](https://github.com/informalsystems/unclog) to manage our changelog)
- [ ] Updated relevant documentation (`docs/` or `spec/`) and code comments

### CODE REVIEW

File: version/version.go [hunk 1]
```diff
@@ -1,9 +1,9 @@
 package version
 
 const (
-	// TMVersionDefault is the used as the fallback version of CometBFT
+	// CMTSemVer is the used as the fallback version of CometBFT
 	// when not using git describe. It is formatted with semantic versioning.
-	TMCoreSemVer = "0.39.0-dev"
+	CMTSemVer = "0.39.0-dev"
```
Comment (thanethomson) [inline version/version.go:6]: These changes in the `version` package need a changelog entry, since they'll be breaking.

**Review by thanethomson — COMMENTED** [review]
There's one more change needed in the `common.mk` file in the root of the repo: https://github.com/cometbft/cometbft/blob/c86d72d29af4f5747045a267e43b00861982a1ca/common.mk#L6 (should be `CMTGitCommitHash`).

If you want to test whether this works, build a binary and the Docker image locally and run the `version` subcommand of the `cometbft` binary and check that the output makes sense.

**Comment by lasarojc** [comment]
> There's one more change needed in the `common.mk` file in the root of the repo: https://github.com/cometbft/cometbft/blob/c86d72d29af4f5747045a267e43b00861982a1ca/common.mk#L6 (should be `CMTGitCommitHash`).

> If you want to test whether this works, build a binary and the Docker image locally and run the `version` subcommand of the `cometbft` binary and check that the output makes sense. 

Thanks for catching. Done
```

./build/cometbft version
0.39.0-dev+c86d72d29
```

**Review by thanethomson — APPROVED** [review]

**Review by melekes — APPROVED** [review]
👍

**Pull request merged**